### PR TITLE
Add .nvmrc and node version checker

### PR DESCRIPTION
**Pull request by bbogdanov**

### CODE REVIEW

File: check-engines-requirements.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var semver = require('semver')
+var packageJson = require('./package')
```
Comment (RubenVerborgh) [inline check-engines-requirements.js]: Let's use `const` for imports.

File: check-engines-requirements.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const nodeVersion = packageJson.engines.node
+
+if (!semver.satisfies(process.version, nodeVersion)) {
+  console.log(
+    `Required node version ${nodeVersion} not satisfied with current version ${
+      process.version
+    }.`
+  )
+  process.exit(1)
+}
```
Comment (RubenVerborgh) [inline check-engines-requirements.js]: We might want the file to be executable and have `#!/usr/bin/env node` at the top.

Comment (RubenVerborgh) [reply check-engines-requirements.js]: Also, let's make this a hidden file, something like `.check-engine-version`.

File: .nvmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+10
```
Comment (RubenVerborgh) [inline .nvmrc]: I don't think we want this in the repo. It's up to individual developers to state their preferences here, no?

Comment (RubenVerborgh) [reply .nvmrc]: Also, .nvmrc [requires a newline](https://github.com/creationix/nvm/tree/v0.33.11#nvmrc).

Comment (bbogdanov) [reply .nvmrc]: Sure it is. This file will just help the maintainers and the contributors to use the same node version via nvm. It is not a necessary file but I think it definitely helps to switch between node version easily.

Comment (RubenVerborgh) [reply .nvmrc]: Yes, but what if I don't want to switch to 10? I might be using 8 or 11.

It seems to me that this file should be there for an individual developer, but in .gitignore so everyone can use their own. And at least we have the script now to warn us if we're using the wrong version? Are there many projects that have .nvmrc in their repository?

**Comment by RubenVerborgh** [comment]
@bbogdanov Shall we just make the checker a hidden file and executable? Then it's all good to go.

**Comment by bbogdanov** [comment]
> @bbogdanov Shall we just make the checker a hidden file and executable? Then it's all good to go.

@RubenVerborgh The file is now a dot file and executable.

**Comment by RubenVerborgh** [comment]
Thanks! is .nvmrc still missing a newline though?

**Comment by RubenVerborgh** [comment]
Thanks a lot!

**Comment by bbogdanov** [comment]
Great, you are welcome!

**Comment by RubenVerborgh** [comment]
The engine check caused problems when installing the package, since the script was being run on `postinstall`. Rather than tweaking this, I removed the script in 663e28fe9307badc1ec5a85772abc2265db550d8. (Have not seen other repositories use it.)

**Comment by bbogdanov** [comment]
Could you give me more information about them/it?

**Comment by RubenVerborgh** [comment]
So when running `npm install` in a repository that has `solid-auth-client` as one of its dependencies, the following problems would occur:
- the script would not be found
- `semver` would not be installed (because it is a `devDependency`)

These issues broke the release of solid-auth-client 2.2.7, which had to be retracted.

While they might be fixable (in multiple ways), I think the fragility is bigger than the utility.

**Comment by bbogdanov** [comment]
I didn't have this in mind. You are right, these problems could occur. Thank you for letting me know!

**Pull request merged**